rpcs3/Emu/RSX/Common/texture_cache.h

```diff
@@ -258,35 +277,80 @@ namespace rsx
 		const u32 m_max_zombie_objects = 64; //Limit on how many texture objects to keep around for reuse after they are invalidated
 
 		//Other statistics
-		const u32 m_cache_miss_threshold = 8; // How many times an address can miss speculative writing before it is considered high priority
-		std::atomic<u32> m_num_flush_requests = { 0 };
-		std::atomic<u32> m_num_cache_misses = { 0 };
-		std::atomic<u32> m_num_cache_speculative_writes = { 0 };
-		std::atomic<u32> m_num_cache_mispredictions = { 0 };
+		std::atomic<u32> m_flushes_this_frame = { 0 };
+		std::atomic<u32> m_misses_this_frame  = { 0 };
+		std::atomic<u32> m_speculations_this_frame = { 0 };
+		static const u32 m_predict_max_flushes_per_frame = 100; // Above this number the predictions are disabled
 
 		// Invalidation
 		static const bool invalidation_ignore_unsynchronized = true; // If true, unsynchronized sections don't get forcefully flushed unless they overlap the fault range
 		static const bool invalidation_keep_ro_during_read = true; // If true, RO sections are not invalidated during read faults
 
 
+
+
 		/**
-		 * Virtual Methods
+		 * Derived Wrappers
```

Comment:
Isn't the whole point of traits to have a traits::method_call() structure? The whole derived()->call thing is just a makeshift man's virtual call impl with none of the benefits of traits afaik. But then again, I'm not very good at c++, so I'll ask for others to help with this.